### PR TITLE
[frontend/backend] EXCLUSION_LIST feature flag removed (#8941)

### PR DESCRIPTION
**Pull request by ValentinBouzinFiligran**
### Proposed changes

* EXCLUSION_LIST feature flag removed

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/8941

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-graphql/src/modules/indicator/indicator-domain.ts [hunk 1]
```diff
@@ -223,17 +223,15 @@ const validateIndicatorPattern = async (context: AuthContext, user: AuthUser, pa
...
-    for (let i = 0; i < observableValues.length; i += 1) {
-      const exclusionListCheck = await checkObservableValue(observableValues[i]);
-      if (exclusionListCheck) {
-        throw FunctionalError(`Indicator of type ${patternType} is contained in exclusion list.`, {
-          doc_code: 'INDICATOR_PATTERN_EXCLUDED',
-          excludedValue: exclusionListCheck.value,
-          exclusionList: exclusionListCheck.listId
-        });
-      }
+  const observableValues = getObservableValuesFromPattern(formattedPattern);
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/modules/indicator/indicator-domain.ts]: I think that we could check if there is something in the cache before parsing observable values. Otherwise we don't need to get observable values. 
We could add a method in exclusionListCache to test if the cache is empty.

File: opencti-platform/opencti-graphql/src/modules/indicator/indicator-domain.ts [hunk 2]
```diff
@@ -223,7 +223,8 @@ const validateIndicatorPattern = async (context: AuthContext, user: AuthUser, pa
   }
 
   // Check that indicator is not excluded from an exclusion list
-  if (isFeatureEnabled('EXCLUSION_LIST')) {
+  // no need to check if the exclusion list cache is empty
+  if (!isCacheEmpty()) {
```
Comment (labo-flg) [inline opencti-platform/opencti-graphql/src/modules/indicator/indicator-domain.ts:227]: @SouadHadjiat is it what you have in mind ?
We avoid calling anything if the cache is empty.

**Comment by CelineSebe** [comment]
Could you rebase please for the CI?

**Review by SouadHadjiat — APPROVED** [review]
